src/backend/gl/src/device.rs

```diff
@@ -1044,24 +1051,36 @@ impl d::Device<B> for Device {
             n::Buffer::Bound { .. } => panic!("Unexpected Buffer::Bound"),
         };
 
-        let (type_mask, alignment) = if usage.contains(buffer::Usage::INDEX) {
-            // Buffers are only allowed to be INDEX usage XOR another type of usage, they are not
-            // allowed to have both INDEX and non-INDEX usage.
-            if !(buffer::Usage::INDEX | buffer::Usage::TRANSFER_SRC | buffer::Usage::TRANSFER_DST).contains(usage) {
-                error!("gl backend does not allow a buffer to be used for both INDEX and non-INDEX usage");
-                (0, 1)
-            } else {
-                // Alignment of 4 covers indexes of type u16 and u32
-                (INDEX_MEM_TYPE_MASK, 4)
-            }
+        let alignment = if usage.contains(buffer::Usage::INDEX) {
```

Comment:
why don't we just return 4 unconditionally here? It's a reasonable alignment to adhere to.

Reply:
Sure thing, that seems reasonable.